### PR TITLE
Submit Tamagotchi 01 assignment

### PR DESCRIPTION
**Pull request by Retamogordo**

### CODE REVIEW

File: frontend/yarn.lock
Comment (shamilsan) [inline frontend/yarn.lock:1]: Unnecessary changes.

File: .github/workflows/jekyll-gh-pages.yml
Comment (shamilsan) [inline .github/workflows/jekyll-gh-pages.yml:1]: What this file is for?

File: 01-tamagotchi/src/lib.rs [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
 
 #[no_mangle]
 extern "C" fn handle() {
-    // TODO: 6️⃣ Add handling of `Name` and `Age` actions
+    let action: TmgAction = msg::load().expect("Error in loading Tamagotchi Action");
+    let tmg = unsafe { TAMAGOTCHI.as_ref().unwrap() };
+
+    let _ = match action {
+        TmgAction::Name => msg::reply(tmg.name.clone(), 0),
+        TmgAction::Age => msg::reply(exec::block_timestamp() - tmg.date_of_birth, 0),
```
Comment (shamilsan) [inline 01-tamagotchi/src/lib.rs:27]: Need to reply with `TmgEvent` enum here.

**Review by shamilsan — CHANGES_REQUESTED** [review]
You are to:

- [X] Fork this repo.
- [ ] Rename it according to the rules: https://academy.gear.rs/getting-started/homework-rules
- [ ] Enable GitHub actions for it.
- [ ] Create a PR for your repo, but not for this one.

**Comment by Retamogordo** [comment]
Hi Shamil,
thank you for your feedback and sorry for the mess.

I'll try to fix it.

-- 
 Sent with Tutanota, enjoy secure & ad-free emails. 



Sep 4, 2023, 05:06 by ***@***.***:

**Comment by shamilsan** [comment]
Hi,
So I am waiting for your PR at https://github.com/Retamogordo/yuryy-gear-academy

Closing this PR.

**Pull request closed without merging**